core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/dao/OrderDaoImpl.java

```diff
@@ -319,27 +319,21 @@ public Order readNamedOrderForCustomer(final Customer customer, final String nam
         query.setParameter("customerId", customer.getId());
         query.setParameter("orderStatus", OrderStatus.NAMED.getType());
         query.setParameter("orderName", name);
-        query.setHint(QueryHints.HINT_CACHEABLE, true);
-        query.setHint(QueryHints.HINT_CACHE_REGION, "query.Order");
-        List<Order> orders = query.getResultList();
-        
-        // Filter out orders that don't match the current locale (if one is set)
         if (BroadleafRequestContext.getBroadleafRequestContext() != null) {
-            ListIterator<Order> iter = orders.listIterator();
-            while (iter.hasNext()) {
-                Locale locale = BroadleafRequestContext.getBroadleafRequestContext().getLocale();
-                Order order = iter.next();
-                if (locale != null && !locale.equals(order.getLocale())) {
-                    iter.remove();
-                }
+            final Locale locale = BroadleafRequestContext.getBroadleafRequestContext().getLocale();
```

Comment:
I think this needs a little bit more, see the named query BC_READ_NAMED_ORDER_FOR_CUSTOMER that is in config/bc/jpa/domain/Order.orm.xml:64
Guess we need to add something like this to that query
AND (:locale is null or order.locale = :locale)

or maybe just create a new named query same as above and accepts locale as parameter and use it in case  BroadleafRequestContext.getBroadleafRequestContext().getLocale() != null ?

Reply:
The Java logic should still be honored.  If there is not a BRC locale, don't filter on locale.  If there is a BRC locale, use it.

Reply:
AND (:locale is null or order.locale = :locale)
This construct provides filtering if locale is present (not null) and ignored if locate is null.

Reply:
I have checked both options (locale = null, and not null) and both work fine.